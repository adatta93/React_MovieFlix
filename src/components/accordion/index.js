import React, { createContext, useContext, useState } from "react";
import {
  Body,
  Container,
  Header,
  Inner,
  Item,
  Title,
} from "./styles/accordion";

const ToggleContext = createContext();

export default function Accordion({ children, ...restProps }) {
  return <Inner {...restProps}>{children}</Inner>;
}

Accordion.Container = function AccordionContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggleShow, setToggleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);
  return (
    <Header onClick={() => setToggleShow((prevVal) => !prevVal)} {...restProps}>
      {children}
      <img
        src="/images/icons/add.png"
        alt="Open"
        className={toggleShow ? "open" : "closed"}
      />
    </Header>
  );
};

Accordion.Body = function AccordionBody({ children, ...restProps }) {
  const { toggleShow } = useContext(ToggleContext);
  return (
    <Body className={toggleShow ? "open" : "closed"} {...restProps}>
      <span>{children}</span>
    </Body>
  );
};
