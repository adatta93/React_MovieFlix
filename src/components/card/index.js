import React, { createContext, useContext, useState } from "react";
import {
  Container,
  Group,
  Title,
  SubTitle,
  Text,
  Item,
  Meta,
  Entities,
  Image,
  FeatureContent,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
} from "./styles/card";

export const CardFeatureContext = createContext();

export default function Card({ children, ...restProps }) {
  const [showCardFeature, setShowCardFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});
  return (
    <CardFeatureContext.Provider
      value={{
        showCardFeature,
        setShowCardFeature,
        itemFeature,
        setItemFeature,
      }}>
      <Container {...restProps}>{children}</Container>
    </CardFeatureContext.Provider>
  );
}

Card.Group = function CardGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Card.Feature = function CardFeatuer({ category, children, ...restProps }) {
  const { showCardFeature, itemFeature, setShowCardFeature } = useContext(
    CardFeatureContext
  );
  return showCardFeature ? (
    <Feature
      {...restProps}
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}>
      <FeatureContent>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowCardFeature(false)}>
          <img src="/images/icons/close.png" alt="Close" />
        </FeatureClose>
        <Group margin="30px 0" flexDirection="row" alignItems="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? "PG" : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold" textTransform="capitalize">
            {itemFeature.genre}
          </FeatureText>
        </Group>
        {children}
      </FeatureContent>
    </Feature>
  ) : null;
};

Card.Item = function CardItem({ item, children, ...restProps }) {
  const { setShowCardFeature, setItemFeature } = useContext(CardFeatureContext);
  return (
    <Item
      onClick={() => {
        setShowCardFeature(true);
        setItemFeature(item);
      }}
      {...restProps}>
      {children}
    </Item>
  );
};

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>;
};

Card.Title = function CardTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Card.SubTitle = function CardSubTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Card.Text = function CardText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Card.Meta = function CardMeta({ children, ...restProps }) {
  return <Meta {...restProps}>{children}</Meta>;
};

Card.Image = function CardImage({ ...restProps }) {
  return <Image {...restProps} />;
};
