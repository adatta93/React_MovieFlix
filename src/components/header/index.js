import React, { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Background,
  ButtonLink,
  Container,
  Highlight,
  HighlightCallOut,
  Logo,
  PlayButton,
  TextLink,
  Picture,
  Group,
  Dropdown,
  Profile,
  Text,
  Search,
  SearchInput,
  SearchIcon,
} from "./styles/header";

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? (
    <Background {...restProps} data-testid="header-bg">
      {children}
    </Background>
  ) : (
    children
  );
}

Header.Container = function HeaderContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ to, children, ...restProps }) {
  return (
    <ButtonLink to={to} {...restProps}>
      {children}
    </ButtonLink>
  );
};

Header.Highlight = function HeaderHighlight({ children, ...restProps }) {
  return <Highlight {...restProps}>{children}</Highlight>;
};

Header.HighlightCallOut = function HeadeHighlightCallOut({
  children,
  ...restProps
}) {
  return <HighlightCallOut {...restProps}>{children}</HighlightCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={src} {...restProps} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <TextLink {...restProps}>{children}</TextLink>;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((prevState) => !prevState)}
        data-testid="search-click">
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
        placeholder="Search..."
        active={searchActive}
        data-testid="search-input"
      />
    </Search>
  );
};
