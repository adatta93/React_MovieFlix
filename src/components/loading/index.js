import React from "react";
import { LockBody, Picture, ReleaseBody, Spinner } from "./styles/loading";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      {src ? (
        <Picture src={`/images/users/${src}.png`} data-testid="loading-pic" />
      ) : null}
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingReleaseBody() {
  return <ReleaseBody />;
};
