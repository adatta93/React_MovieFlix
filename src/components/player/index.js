import React, { createContext, useContext, useState } from "react";
import ReactDOM from "react-dom";
import { Container, Overlay, Inner, Button, VideoClose } from "./player";

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);
  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);
  const handleClose = (e) => {
    e.stopPropagation();
    e.nativeEvent.stopImmediatePropagation();
    //console.log(e.target, e.currentTarget);
    if (e.target === e.currentTarget) {
      setShowPlayer(false);
    }
  };
  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay {...restProps} onClick={handleClose} data-testid="player">
          <Inner>
            <video id="movieflix-player" controls>
              <source src={src} type="video/mp4" />
            </video>
            <VideoClose
              onClick={() => setShowPlayer(false)}
              data-testid="playerClose">
              <img src="/images/icons/close.png" alt="Close" />
            </VideoClose>
            {/* <Close /> */}
          </Inner>
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  const { setShowPlayer } = useContext(PlayerContext);
  return (
    <Button {...restProps} onClick={() => setShowPlayer((prev) => !prev)}>
      Play
    </Button>
  );
};
