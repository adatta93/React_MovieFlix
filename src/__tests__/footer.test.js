import React from "react";
import { render } from "@testing-library/react";
import { Footer } from "../components";

describe("Footer", () => {
  it("renders the Footer with data", () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Call 000-100-200-1234</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Releations</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help Centre</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Cookie Preferences</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Ways to Watch</Footer.Link>
            <Footer.Link href="#">Corporate Information</Footer.Link>
            <Footer.Link href="#">MovieFlix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Media Centre</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>MovieFlix India</Footer.Text>
      </Footer>
    );

    expect(getByText("Questions? Call 000-100-200-1234")).toBeTruthy();
    expect(getByText("FAQ")).toBeTruthy();
    expect(getByText("Help Centre")).toBeTruthy();
    expect(getByText("Account")).toBeTruthy();
    expect(getByText("Media Centre")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
