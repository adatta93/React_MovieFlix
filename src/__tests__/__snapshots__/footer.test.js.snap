// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the Footer with data 1`] = `
<div
  class="footer__Container-sc-1i7nybi-0 jWWrkW"
>
  <p
    class="footer__Title-sc-1i7nybi-4 geqkpz"
  >
    Questions? Call 000-100-200-1234
  </p>
  <div
    class="footer__Break-sc-1i7nybi-6 kNcXVU"
  />
  <div
    class="footer__Row-sc-1i7nybi-2 klKYVe"
  >
    <div
      class="footer__Column-sc-1i7nybi-1 fpWtyC"
    >
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Investor Releations
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-sc-1i7nybi-1 fpWtyC"
    >
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Legal Notices
      </a>
    </div>
    <div
      class="footer__Column-sc-1i7nybi-1 fpWtyC"
    >
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        MovieFlix Originals
      </a>
    </div>
    <div
      class="footer__Column-sc-1i7nybi-1 fpWtyC"
    >
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Media Centre
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-sc-1i7nybi-3 kFBBlZ"
        href="#"
      >
        Contact Us
      </a>
    </div>
  </div>
  <div
    class="footer__Break-sc-1i7nybi-6 kNcXVU"
  />
  <p
    class="footer__Text-sc-1i7nybi-5 cuaPMt"
  >
    MovieFlix India
  </p>
</div>
`;
